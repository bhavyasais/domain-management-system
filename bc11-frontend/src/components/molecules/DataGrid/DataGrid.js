import React from "react";
import { AgGridReact } from "ag-grid-react";

import "ag-grid-community/dist/styles/ag-grid.css";
import "ag-grid-community/dist/styles/ag-theme-balham.css";

export default function DataGrid({ columnDefs, rowData }) {
  return (
    <div
      className="ag-theme-balham"
      style={{
        width: 1204,
        height: 500,
      }}
    >
      <AgGridReact
        columnDefs={columnDefs}
        rowData={rowData}
        rowSelection="multiple"
      />
    </div>
  );
}
