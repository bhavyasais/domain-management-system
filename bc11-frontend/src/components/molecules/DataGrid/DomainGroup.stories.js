import React from "react";
import {
  DomainrowData,
  DomaincolumnDefs,
} from "../../../utils/commandCentermockData";
import DataGrid from "./DataGrid";

export default {
  title: "Molecules/Grid",
};

export const domainGroup = () => {
  return <DataGrid columnDefs={DomaincolumnDefs} rowData={DomainrowData} />;
};
