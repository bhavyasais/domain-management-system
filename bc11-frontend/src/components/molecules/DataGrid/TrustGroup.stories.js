import React from "react";
import DataGrid from "./DataGrid";

export default {
  title: "Molecules/Grid",
};

const clickedCheckbox = (params) => {
}

const checkboxRenderer = (params) => {
  return <div><input type="checkbox" onClick={() => clickedCheckbox(params)} /></div>;
}

const headerRenderer = (params) => {
  const name = params.displayName;
  return <div><input type="checkbox" onClick={() => clickedCheckbox(params)} /><label>{name}</label></div>;
}

const redoFunction = (headerName, headerField) => {
  return  { headerName: headerName, field: headerField, cellRendererFramework: (params) => {
    return checkboxRenderer(params);
  }, headerComponentFramework: (params) => {
    return headerRenderer(params);
  }, cellStyle: {textAlign: 'center'}};

}

const columnDefs = [
    { headerName: "S no", valueGetter: "node.rowIndex + 1"},
    { headerName: 'A Trust Group', field: 'trustGroup', sortable: true, filter: true },
];

columnDefs.push(redoFunction("Edit File?", "editFile"));
columnDefs.push(redoFunction("View File?", "viewFile"));
columnDefs.push(redoFunction("Delete File?", "deleteFile"));
columnDefs.push(redoFunction("Share File?", "shareFile"));

const rowData = [
    { trustGroup: 'DO NOT TRUST'},
    { trustGroup: 'Top 10'},
    { trustGroup: 'Top 10000'},
    { trustGroup: 'Top 30000'}
];

export const trustGroup = () => {
  return <DataGrid columnDefs={columnDefs} rowData={rowData}/>;
};
