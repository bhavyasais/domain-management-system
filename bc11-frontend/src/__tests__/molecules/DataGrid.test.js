import React from "react";
import { mount } from "enzyme";
import DataGrid from "../../components/molecules/DataGrid/DataGrid";
import {
  DomaincolumnDefs,
  DomainrowData,
} from "../../utils/commandCentermockData";

describe("<DataGrid/>", () => {
  it("Renders correctly", () => {
    let wrapper;
    wrapper = mount(
      <DataGrid rowData={DomainrowData} columnDefs={DomaincolumnDefs} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
