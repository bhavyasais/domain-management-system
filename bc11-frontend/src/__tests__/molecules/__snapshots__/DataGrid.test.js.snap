// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataGrid/> Renders correctly 1`] = `ReactWrapper {}`;
